mne/preprocessing/annotate_amplitude.py

```diff
@@ -0,0 +1,227 @@
+# Author: Mathieu Scheltienne <mathieu.scheltienne@fcbg.ch>
+#
+# License: BSD-3-Clause
+
+import numpy as np
+
+from ..fixes import jit
+from ..io import BaseRaw
+from ..annotations import (Annotations, _adjust_onset_meas_date,
+                           _annotations_starts_stops)
+from ..io.pick import _picks_to_idx, _picks_by_type, _get_channel_types
+from ..utils import _validate_type, verbose, logger, _mask_to_onsets_offsets
+
+
+@verbose
+def annotate_amplitude(raw, peak=None, flat=None, bad_percent=5,
+                       min_duration=0.005, picks=None, *, verbose=None):
+    """Annotate segments of raw data based on PTP amplitude.
+
+    Annotate segments of raw data which PTP amplitudes between consecutive
+    samples exceeds thresholds in ``peak`` or fall below thresholds in
+    ``flat``.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+    peak : float | dict | None
+        Reject segments based on **maximum** peak-to-peak signal amplitude
+        (PTP). Valid **keys** can be any channel type present in the object.
+        The **values** are floats that set the minimum acceptable PTP. If the
+        PTP is smaller than this threshold, the segment will be annotated.
+        If float, the minimum acceptable PTP is applied to all channels.
+    flat : float | dict | None
+        Reject segments based on **minimum** peak-to-peak signal amplitude
+        (PTP). Valid **keys** can be any channel type present in the object.
+        The **values** are floats that set the minimum acceptable PTP. If the
+        PTP is smaller than this threshold, the segment will be annotated.
+        If float, the minimum acceptable PTP is applied to all channels.
+    bad_percent : float
+        The percentage of the time a channel can be bad.
+        Below this percentage, temporal bad marking (:class:`~mne.Annotations`)
+        will be used. Above this percentage, spatial bad marking
+        (:class:`info['bads'] <mne.Info>`) will be used.
```

Comment:
maybe I'm not looking closely enough at the code, but you're not actually marking the channels as bad, right?  Just returning the indices?  If that's correct, I think it's misleading here to say that "spatial bad marking will be used"

Reply:
Agree, I'll update.